src/map/unit.cpp

```diff
@@ -780,6 +780,10 @@ int unit_walktoxy( struct block_list *bl, short x, short y, unsigned char flag)
 	}
 
 	TBL_PC *sd = BL_CAST(BL_PC, bl);
+	if (sd && bl->type == BL_PC) {
```

Comment:
```suggestion

	if (sd) {
```
`sd` is a BL_PC automatically.